tests/Unit/ActionTesting.hpp

```diff
@@ -324,24 +491,37 @@ class MockArrayElementProxy {
     tuples::get<InboxTag>(inbox_)[id].emplace(data);
   }
 
-  MockLocalAlgorithm<Component>* ckLocal() { return &local_algorithm_; }
+  template <typename Action, typename... Args>
+  void simple_action(std::tuple<Args...> args) noexcept {
+    local_algorithm_.template simple_action<Action>(std::move(args));
+  }
+
+  template <typename Action>
+  void simple_action() noexcept {
+    local_algorithm_.template simple_action<Action>();
+  }
+
+  MockDistributedObject<Component>* ckLocal() { return &local_algorithm_; }
 
  private:
-  MockLocalAlgorithm<Component>& local_algorithm_;
+  MockDistributedObject<Component>& local_algorithm_;
   Inbox& inbox_;
 };
 
+// A mock class for the Charm++ generated CProxy_AlgorithmArray (we use an array
+// for everything, so no need to mock groups, nodegroups, singletons).
 template <typename Component, typename Index, typename InboxTagList>
 class MockProxy {
  public:
   using Inboxes =
-      std::unordered_map<Index, tuples::TaggedTupleTypelist<InboxTagList>>;
-  using LocalAlgorithms =
-      std::unordered_map<Index, MockLocalAlgorithm<Component>>;
+      std::unordered_map<Index,
+                         tuples::tagged_tuple_from_typelist<InboxTagList>>;
+  using TupleOfMockDistributedObjects =
+      std::unordered_map<Index, MockDistributedObject<Component>>;
 
   MockProxy() : inboxes_(nullptr) {}
 
-  void set_data(LocalAlgorithms* local_algorithms,
+  void set_data(TupleOfMockDistributedObjects* local_algorithms,
```

Comment:
The commit message for the commit introducing this change has one too many "Mock"s in it.